### ISSUE TITLE
bug report: Team id and Channel id are same at channelData property from context on Teams-chat data.

### ISSUE THREAD

**Issue by tuxxon** [OfficeDev/TeamsFx#7341]
**Describe the bug**
Team.id and Channel.id are the same at channelData property from the context parameter on Teams-chat data.
Team.id and Channel.id must be different as you see on MS-ChangeNotifiction.
But, the bot-app shows the same value for team Id and Channel Id.

**To Reproduce**
1. Running the existing message extension.
2. $ teamsfx provision
3. $ teamsfx deploy
4. $ teamsfx package
5. open teamsApp 
6. click `Apps`.
7. click `Manage your app`.
8. click `Publish an app`.
9. click `Upload a custom app`.
10. click more (...) button on the top right of message.
11. select a menu I created.

**Expected behavior**
Team id and Channel id must be different as you see on MS-ChangeNotifiction as follows:
```
"@odata.id": "teams('eff642a9-7be2-4b8e-9c10-b25ba014e589')/channels('19:MyWlSsl4r8lvCsgeJm2_gz3xTHbnVHnPXDD2C8CFnTw1@thread.tacv2')/messages('1667636950355')/replies('1667651318633')"
```
or
```
    "channelIdentity": {
      "channelId": "19:MyWlSsl4r8lvCsgeJm2_gz3xTHbnVHnPXDD2C8CFnTw1@thread.tacv2",
      "teamId": "eff642a9-7be2-4b8e-9c10-b25ba014e589"
    },
```
But, Bot app shows the same value for team.Id and channel.Id
```
    "channelData": {
      "channel": {
        "id": "19:XybrwPUGgGoJdX8YDXp2pM5aZ6tZP-9dyhL_nyLOHB41@thread.tacv2"
      },
      "team": {
        "id": "19:XybrwPUGgGoJdX8YDXp2pM5aZ6tZP-9dyhL_nyLOHB41@thread.tacv2"
      },
      "tenant": {
        "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      },
      "source": {
        "name": "compose"
      }
    },
```

**Screenshots**
N/A

**VS Code Extension Information (please complete the following information):**
 - OS: WSL2-ubuntu22- Windows 10, 
 - Version 4.2.1

**CLI Information (please complete the following information):**
 - OS: windows 10, chrome - 108.0.5359.125, 64bit
 - Version:  [teams](https://teams.microsoft.com/)

**Additional context**
- You can get my basic source getting the same values.
https://github.com/OfficeDev/TeamsFx/issues/7276

- Posted question on stack overflow
   https://stackoverflow.com/questions/74928566/can-i-get-teamid-from-the-context-of-message-extention-on-teams-microsoft-com

**Comment by tuxxon** [comment]
Hello @adashen 

I've added the debug codes as follows:
```
  public async handleTeamsMessagingExtensionFetchTask(
    context: TurnContext,
    action: any
  ): Promise<any> {

  console.log(`[DEBUG] handleTeamsMessagingExtensionFetchTask()`);    
  const teamDetails = await TeamsInfo.getTeamDetails(context);
  console.log(`[DEBUG] teamDetails = ${teamDetails}`);

    switch (action.commandId) {
    case 'OpenAssessments':
        return openAssessments(context, action);
    case 'OpenAdvice':
        return await openAdvice(context, action);
    case 'ClosedAssessment':
        return await closedAssessment(context, action);
    default:
      throw new Error("NotImplemented");  
  }
```

And, I've got error messages below:
```
[DEBUG] handleTeamsMessagingExtensionFetchTask()

 [onTurnError] unhandled error: Error: This method is only valid within the scope of a MS Teams Team.
/home/tuxxon/thepsyentist/jeisys/pas.teamsplugin/bot/node_modules/@azure/ms-rest-js/lib/restError.ts:25
    super(message);
    ^
Error: BotFrameworkAdapter.processActivity(): 500 ERROR
 Error: The bot is not part of the conversation roster.
    at new RestError (/home/tuxxon/thepsyentist/jeisys/pas.teamsplugin/bot/node_modules/@azure/ms-rest-js/lib/restError.ts:25:5)
    at /home/tuxxon/thepsyentist/jeisys/pas.teamsplugin/bot/node_modules/@azure/ms-rest-js/lib/policies/deserializationPolicy.ts:150:27
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at BotFrameworkAdapter.<anonymous> (/home/tuxxon/thepsyentist/jeisys/pas.teamsplugin/bot/node_modules/botbuilder/src/botFrameworkAdapter.ts:1273:23)
    at Generator.throw (<anonymous>)
    at rejected (/home/tuxxon/thepsyentist/jeisys/pas.teamsplugin/bot/node_modules/botbuilder/lib/botFrameworkAdapter.js:13:65)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Waiting for the debugger to disconnect...
[nodemon] app crashed - waiting for file changes before starting...
```
![image](https://user-images.githubusercontent.com/7301022/209756890-82fcf585-e7d2-4bfc-8c88-af896102be4e.png)

**Comment by tuxxon** [comment]
Please let me knw how to get teamId or how to fix this problem above.

**Comment by adashen** [comment]
Hi @tuxxon can you provide more information like how you install the bot? 
1) a "1-1" 2) a "group chat", where there's a direct chat between, for example, you, one or more other users, and the bot (this will also appear in the "chat" section on the left menu in Teams) 3) A "channel" inside a "Team" in Teams. 
Is it in 1-1 chat or in a group chat? From the error, it seems the app is not in group

**Comment by tuxxon** [comment]
Hello @adashen 
I've got the same error from both conversationType = 'groupChat' and 'channel' as above.

Please let me know how to fix this issue.

**Comment by tuxxon** [comment]
In my opinion, 
You can create a sample app from Teams Toolkit - message extension.
Then, After adding my method to the sample app, 
just run it, I think you can get the same error 100%.

**Comment by adashen** [comment]
Hi, @tuxxon, please add the following block in to the teams app manifest. (templates/appPackage/manifest.template.json in our default messaging extension template) 
"bots": [
        {
            "botId": "{{state.fx-resource-bot.botId}}",
            "scopes": [
                "team",
                "personal",
                "groupchat"
            ],
            "needsChannelSelector": false,
            "isNotificationOnly": false,
            "supportsFiles": true,
            "supportsCalling": false,
            "supportsVideo": true
        }
    ],
The configuration enables the "bot capability" of the teams app in "team, personal, groupchat" scopes. When F5 to lauch the app and teams pop up the installation page. Please click the arrow besides the "Add" button, and choose "team" to add it. After that, go to the team installed the app and try the messaging extension. In this way, the "bot" is added to the team, so it could get the team context.
![image](https://user-images.githubusercontent.com/1872137/209895898-8466fc13-bfe2-4525-a000-f68ba73673e7.png)

**Comment by tuxxon** [comment]
@adashen Have a good day.
Thanks a lot. It works very well
